### PR TITLE
 Add ServerVulnerabilityAssessment and ManagedInstanceVulnerabilityAssessment operations

### PR DESCRIPTION
**Pull request by yaakoviyun**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by yaakoviyun** [comment]
REST API PR -https://github.com/Azure/azure-rest-api-specs/pull/4416

**Comment by yaakoviyun** [comment]
@jaredmoo   @yoavfr  @talhers

**Comment by jaredmoo** [comment]
There are test failures, please look at Travis CI build and scroll all the way to the bottom.

I noticed that `ManagedInstanceTestFixture` is reusing the same test recording file for all tests that use it (notice `.ctor.json`). This seems like a problem, the tests that call into `ManagedInstanceTestFixture` should pass in the suite object and method name so that when `ManagedInstanceTestFixture` creates `SqlManagementTestContext` this `suiteObject` and `testName` should be the test method's class and method name. I am not sure however if this is specifically what is causing the test failure.

**Comment by jaredmoo** [comment]
Other than this test issue the change looks good though :)

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part, please regenerate the code once REST spec PR is merged

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part, some unused code needs to be cleaned up

**Comment by dsgouda** [comment]
@yaakoviyun CI tests still failing, please take a look

**Comment by yaakoviyun** [comment]
@dsgouda , @jaredmoo  - tests are passing

**Comment by yaakoviyun** [comment]
@talhers

**Comment by yaakoviyun** [comment]
Ping...

File: src/SDKs/SqlManagement/Sql.Tests/ManagedDatabaseVulnerabilityAssessmentTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿
+namespace Sql.Tests
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/ManagedDatabaseVulnerabilityAssessmentTestFixture.cs]: Is it used in the tests

File: src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceVulnerabilityAssessmentTestFixture.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿
+namespace Sql.Tests
+{
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceVulnerabilityAssessmentTestFixture.cs]: Is it used in the tests

**Comment by dsgouda** [comment]
@yaakoviyun @jaredmoo What Nuget package version do you wish to release this as, `1.24.-0-preview` or `1.25.0-preview`

**Review by jaredmoo — APPROVED** [review]
We should try to release this and the MI collation change together in 1.24. Please make sure you and Nemanja are in sync with each other.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@yaakoviyun please squash commits into a single commit and we are good to merge

**Comment by yaakoviyun** [comment]
How can I do it? What is the method you guys use for squashing?

Sent from Outlook
________________________________
מאת: Deepak Shankargouda <notifications@github.com>
‏‏נשלח: יום רביעי 21 נובמבר 2018 19:36:54
‏‏אל: Azure/azure-sdk-for-net
עותק: Yaakov Iyun; Mention
‏‏נושא: Re: [Azure/azure-sdk-for-net] Add ServerVulnerabilityAssessment and ManagedInstanceVulnerabilityAssessment operations (#5005)


@yaakoviyun<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fyaakoviyun&data=02%7C01%7Cyaiyun%40microsoft.com%7Cd084f053e1a346a5986808d64fd7eda1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636784186157830291&sdata=2t%2BaEZV0PSQwvSalix7LgjrORkY07nsfugkc%2F3RP3Nw%3D&reserved=0> please squash commits into a single commit and we are good to merge

**Comment by dsgouda** [comment]
@yaakoviyun The easiest way is to copy your changes to a temp directory, recreate a branch from latest psSdkJson6 and copy over the changes to this branch.

**Comment by jaredmoo** [comment]
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by jaredmoo** [comment]
So, `git rebase -i psSdkJson6` and then change all "pick" to "squash" except the first one. This will redo your change history on top of `psSdkJson6` branch and squash them down into one commit.

Or, 
```
git reset --hard psSdkJson6
git merge --squash ee511584aa30b93a7032b9afa896a17b40cc11ec
```
This will reset your branch to the same state as `psSdkJson6`, then squash merge all changes up until `ee511584aa30b93a7032b9afa896a17b40cc11ec` into this branch. (This is the commit id of the latest commit in your branch - https://github.com/Azure/azure-sdk-for-net/pull/5005/commits/ee511584aa30b93a7032b9afa896a17b40cc11ec )

These two methods are equivalent, just different UX.

**Comment by jaredmoo** [comment]
@dsgouda Is it possible for you to squash merge in github UI?

**Comment by jaredmoo** [comment]
Oh btw @yaakoviyun after you do this squash you will need to `git push --force` since it rewrites commit history so it is destructive to the remote branch that you are pushing to

**Comment by dsgouda** [comment]
When you do a pull from upstream, the best way to squash commits is as I mentioned above

**Comment by yaakoviyun** [comment]
Will it require a new PR?

I must say that is operation seems very messy and doesn’t feel like a best practice move.

Sent from Outlook
________________________________
מאת: Deepak Shankargouda <notifications@github.com>
‏‏נשלח: יום רביעי 21 נובמבר 2018 20:17:07
‏‏אל: Azure/azure-sdk-for-net
עותק: Yaakov Iyun; Mention
‏‏נושא: Re: [Azure/azure-sdk-for-net] Add ServerVulnerabilityAssessment and ManagedInstanceVulnerabilityAssessment operations (#5005)


When you do a pull from upstream, the best way to squash commits is as I mentioned above

**Comment by yaakoviyun** [comment]
It seems not the right thing to do.
Creating a new PR will hide all the original PR’s comments.
It seems that for every change, a developer will need two PRs and additional manual steps which raises the chance for mistakes.

Is this the way all PRs are managed?

Sent from Outlook
________________________________
מאת: Deepak Shankargouda <notifications@github.com>
‏‏נשלח: יום רביעי 21 נובמבר 2018 20:26:52
‏‏אל: Azure/azure-sdk-for-net
עותק: Yaakov Iyun; Mention
‏‏נושא: Re: [Azure/azure-sdk-for-net] Add ServerVulnerabilityAssessment and ManagedInstanceVulnerabilityAssessment operations (#5005)


@jaredmoo<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjaredmoo&data=02%7C01%7Cyaiyun%40microsoft.com%7C56fd8f0699824f69e42308d64fdee8a9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636784216137842483&sdata=ZcmpTCoXKhyGHQelZHxFpq74ahYbJXXeq8jMZM0T5%2B8%3D&reserved=0> problem is if the "merge" is a part of the commit history, the whole commit history gets messed up too.
As for the concerns about merging, the idea is to make sure the branch is ready to be merged to psSdkJson6 and then recreate a new branch with the ready to merge changes.
If it still feels like an issue, simply create another branch and another PR with it referencing this one so we skip the review.
Hope this helps

**Comment by dsgouda** [comment]
The additional steps are needed only if the branch was updated with remote master and some conflicts had to be resolved, else a simple git rebase should work.
If you are able to do a git rebase successfully, we can proceed with this PR
Apologize for the confusion

**Pull request merged**